src/components/Popups/TransactionPopup.tsx

```diff
@@ -22,22 +24,29 @@ function TransactionPopupContent({ tx, chainId }: { tx: TransactionDetails; chai
   const tokens = useCombinedActiveList()
   const activity = parseLocalActivity(tx, chainId, tokens)
   const { ENSName } = useENSName(activity?.otherAccount)
+  const theme = useTheme()
 
   if (!activity) return null
 
   const explorerUrl = getExplorerLink(chainId, tx.hash, ExplorerDataType.TRANSACTION)
 
+  const currencies = activity.currencies?.filter((c) => c) as Currency[]
```

Comment:
nit: I'm not sure if these will work in tandem, but you can use `Boolean` as an identity filter, and you can usually (depending on your filterFn) type the return type instead of coercing it, which I find to be safer.

```suggestion
  const currencies = activity.currencies?.filter<Currency>(Boolean)
```

Reply:
i believe the type casting for `Currency[]` here is also wrong because `activity.currencies` can be `undefined`

Reply:
not relevant anymore because we can just pass arrays with undefined elements to `PortfolioLogo`